queries/elixir/highlights.scm

```diff
@@ -11,144 +16,184 @@
+  ">="
+  ">>>"
+  "@"
+  "^"
+  "and"
+  "in"
+  "not"
+  "or"
+  "when"
+  "|"
+  "|>"
+  "||"
+  "|||"
+  "~>"
+  "~>>"
+  "~~~"
+  "~"
+] @operator
+
+; Do/Fn blocks
 [
-  "after"
-  "and" 
-  "catch"
   "do"
-  "else"
   "end"
   "fn"
-  "in"
-  "not in"
-  "not"
-  "or"
-  "rescue"
-  "when" 
 ] @keyword
 
+; Exception blocks
 [
-  (comment) 
-  (unused_identifier)
-] @comment
-
-[
-  (heredoc_content)
-  (sigil_content)
-  (string)
-] @string
-
-; __MODULE__ and friends
-(special_identifier) @constant.builtin
-
-(map ["%{" "}"] @constructor)
+  "catch"
+  "rescue" 
+] @exception
 
-(struct ["%" "{" "}"] @constructor)
+; Conditional blocks
+[ 
+  "else" 
+] @conditional
 
-(binary_op operator: _ @operator)
+; Comments
+(comment) @comment
 
-(unary_op operator: _ @operator)
+; Parser errors
+(ERROR) @error
 
-(atom) @symbol
+; Sigils
+(sigil (sigil_name) @operator quoted_start: _ @operator quoted_end: _ @operator)
 
-(keyword) @parameter
+; Modules
+(alias) @type
 
+; Atoms
 [
-  (true) 
-  (false)
-] @boolean
-
-(nil) @constant.builtin
+  (atom)
+  (quoted_atom) 
+] @symbol
 
-(sigil) @string.special
+; Keywords
+(keyword) @symbol
 
-(identifier) @variable
+; Strings
+(string) @string
 
-(module) @type
+; Interpolation
+(interpolation "#{" @string.escape (_)? "}" @string.escape)
 
-(function_identifier) @function
+; Escape sequences
+(escape_sequence) @string.escape
 
+; Integers
 (integer) @number
 
+; Floats
 (float) @float
 
+; Characters
 [
-  (sigil_start) 
-  (sigil_end)
-] @string.special
-
-; Module attributes as "attributes"
-(unary_op operator: "@" @attribute [
-  (call function: (function_identifier) @attribute) 
-  (identifier) @attribute
+  (char)
+  (charlist)
+] @character
+
+; Booleans
+(boolean) @boolean
+
+; Nil
+(nil) @constant.builtin
+
+; Calling a local function
+(call target: (identifier) @function) 
+
+; Calling a remote function
+(call target: (dot left: [
+  (atom) @type 
+  (_)
+] right: (identifier) @function) (arguments))
+
+; Module attributes
+(unary_operator operator: "@" @constant operand: [
+  (identifier) @constant
+  (call target: (identifier) @constant)
 ])
 
-; Erlang modules (when they are the remote of a function call) are highlighted as Elixir modules
-(dot_call remote: (atom) @type)
-
-(call (function_identifier) @keyword.function (#any-of? @keyword.function 
-  "def"
-  "defdelegate"
-  "defexception"
-  "defguard"
-  "defguardp"
-  "defimpl"
-  "defmacro"
-  "defmacrop"
-  "defmodule"
-  "defoverridable"
-  "defp"
-  "defprotocol"
-  "defstruct"
-) [(identifier) @function (_)]) ; 0-arity function def without parens
-
-(call (function_identifier) @include (#any-of? @include 
-  "alias" 
-  "import" 
-  "require" 
-  "use"
+; Unused identifiers
+((identifier) @comment (#match? @comment 
+  "^_([^_].*)?$"
 ))
-
-(call (function_identifier) @conditional (#any-of? @conditional 
+ 
+; Calling a local def function
+(call target: ((identifier) @keyword.function (#match? @keyword.function
+  "^def(.*)?$"
```

Comment:
I was wondering if we should match on the prefix, in the context of `defn` and `defnp` it's useful to have it something generic like this, as anyone can add their own marcos. However, this can very well have false positives, like `default_state()`. It could be slightly better as `"^def([^_]*)?$"`, since the def macros are generally a single word. No strong opinion here though :)